src/components/TradeWidget/index.tsx

```diff
@@ -64,17 +81,37 @@ const TradeWidget: React.FC = () => {
   const tokens = useMemo(() => tokenListApi.getTokens(fallBackNetworkId), [fallBackNetworkId])
 
   // Listen on manual changes to URL search query
-  const { sell: sellTokenSymbol, receive: receiveTokenSymbol } = useParams()
+  const { sell: sellTokenSymbol, buy: receiveTokenSymbol } = useParams()
+  const query = useQuery()
+  const sellAmount = query.get('sell')
+  const receiveAmount = query.get('buy')
 
   const [sellToken, setSellToken] = useState(
     () => getToken('symbol', sellTokenSymbol, tokens) || getToken('symbol', 'DAI', tokens),
   )
   const [receiveToken, setReceiveToken] = useState(
     () => getToken('symbol', receiveTokenSymbol, tokens) || getToken('symbol', 'USDC', tokens),
   )
+  const sellInputId = 'sellToken'
+  const receiveInputId = 'receiveToken'
 
-  // Change URL on internal token change
-  useURLParams(`sell=${sellToken.symbol}-0&receive=${receiveToken.symbol}-20`)
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const sanitizedSellAmount = useMemo(() => sanitizeInputAmount(sellAmount, sellToken.decimals), [])
```

Comment:
What do useMemo without dependencies? shouldn't this depend on the `sellAmount, sellToken.decimals`?
Also, is only for optimizing? Is it really worth it in this case?

Reply:
Yes, for optimization. I had a comment explaining this but I guess during refactoring I got rid of it.

The point of this sanitization is to prevent invalid numbers to be inserted into the input, when coming from user input url. For example: `/trade/USDC-DAI?sell=kjasrhsker&buy=1.1.1`will be converted to `/trade/USDC-DAI?sell=0&buy=0`

Non-numeric values are prevented to be inserted in the input, but the values coming from the url are used as default values, thus bypassing validation.
For this reason, the sanitization is ran only once (the `[]` dependency). Subsequent updates will be validated by the form and we can trust the value won't contain weird things.